pom.xml

```diff
@@ -58,6 +58,11 @@
   </pluginRepositories>
 
   <dependencies>
+    <dependency>
+      <groupId>org.jenkins-ci.plugins</groupId>
+      <artifactId>command-launcher</artifactId>
```

Comment:
Does this plugin actually use the `CommandLauncher` API? I don't see any uses from some quick searches, so maybe the dependency should just be deleted?

Reply:
Lemme check, but I definitely added it after seeing a failure.

Reply:
OK, so confirmed it is needed and fails immediately during tests without it, but AIUI, this is a runtime dependency for during tests. So good catch: modifying this as `test` scope to avoid pulling this as a hard dependency for users.

Reply:
FTR

```
[INFO] Running InjectedTest
[ERROR] Tests run: 5, Failures: 0, Errors: 5, Skipped: 0, Time elapsed: 0.057 s <<< FAILURE! - in InjectedTest
[ERROR] initializationError(InjectedTest)  Time elapsed: 0.009 s  <<< ERROR!
java.lang.NoClassDefFoundError: hudson/slaves/CommandLauncher
        at org.jvnet.hudson.test.JellyTestSuiteBuilder$JellyTestSuite.<init>(JellyTestSuiteBuilder.java:139)
        at org.jvnet.hudson.test.JellyTestSuiteBuilder$JellyTestSuite.<init>(JellyTestSuiteBuilder.java:138)
        at org.jvnet.hudson.test.JellyTestSuiteBuilder.build(JellyTestSuiteBuilder.java:80)
        at org.jvnet.hudson.test.PluginAutomaticTestBuilder.build(PluginAutomaticTestBuilder.java:59)
        at InjectedTest.suite(InjectedTest.java:17)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
        at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.base/java.lang.reflect.Method.invoke(Method.java:566)
        at org.junit.internal.runners.SuiteMethod.testFromSuiteMethod(SuiteMethod.java:35)
        at org.junit.internal.runners.SuiteMethod.<init>(SuiteMethod.java:24)
        at org.junit.internal.builders.SuiteMethodBuilder.runnerForClass(SuiteMethodBuilder.java:11)
        at org.junit.runners.model.RunnerBuilder.safeRunnerForClass(RunnerBuilder.java:59)
        at org.junit.internal.builders.AllDefaultPossibilitiesBuilder.runnerForClass(AllDefaultPossibilitiesBuilder.java:26)
        at org.junit.runners.model.RunnerBuilder.safeRunnerForClass(RunnerBuilder.java:59)
        at org.junit.internal.requests.ClassRequest.getRunner(ClassRequest.java:33)
        at org.apache.maven.surefire.junit4.JUnit4Provider.execute(JUnit4Provider.java:362)
        at org.apache.maven.surefire.junit4.JUnit4Provider.executeWithRerun(JUnit4Provider.java:273)
        at org.apache.maven.surefire.junit4.JUnit4Provider.executeTestSet(JUnit4Provider.java:238)
        at org.apache.maven.surefire.junit4.JUnit4Provider.invoke(JUnit4Provider.java:159)
        at org.apache.maven.surefire.booter.ForkedBooter.invokeProviderInSameClassLoader(ForkedBooter.java:384)
        at org.apache.maven.surefire.booter.ForkedBooter.runSuitesInProcess(ForkedBooter.java:345)
        at org.apache.maven.surefire.booter.ForkedBooter.execute(ForkedBooter.java:126)
        at org.apache.maven.surefire.booter.ForkedBooter.main(ForkedBooter.java:418)
Caused by: java.lang.ClassNotFoundException: hudson.slaves.CommandLauncher
        at java.base/jdk.internal.loader.BuiltinClassLoader.loadClass(BuiltinClassLoader.java:583)
        at java.base/jdk.internal.loader.ClassLoaders$AppClassLoader.loadClass(ClassLoaders.java:178)
        at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:521)
        ... 24 more
```

Reply:
Done in https://github.com/jenkinsci/display-url-api-plugin/pull/24/commits/4e09683df527f32ce08d4cf352385249c44fe423

Reply:
Ah, I think it's because of the old explicit jenkins-test-harness dependency. Remove that line so you pick up the version from the parent pom (which should be new enough to include https://github.com/jenkinsci/jenkins-test-harness#231-2017-oct-17) and I think you'd be able to delete the dependency.

Reply:
Absoloooltely. Bumped and removed inhttps://github.com/jenkinsci/display-url-api-plugin/pull/24/commits/021390dc07854861e7e2616212c3164583a4a7a8, great point!